emmet-core/emmet/core/search.py

```diff
@@ -261,3 +261,133 @@ class SearchDoc(PropertyDoc):
     # Theoretical
 
     theoretical: bool = Field(True, description="Whether the material is theoretical.")
+
+    @classmethod
+    def from_docs(cls, material_id: MPID, **docs: Dict[str, Dict]):
+        """Converts a bunch of search docs into a SearchDoc"""
+        doc = _copy_from_doc(docs)
+
+        # Reshape document for various sub-sections
+        # Electronic Structure + Bandstructure + DOS
+        if "bandstructure" in doc:
+            doc["has_props"].append("bandstructure")
+        if "dos" in doc:
+            doc["has_props"].append("dos")
+        if "calc_id" in doc:
+            doc["es_source_calc_id"] = doc["calc_id"]
+            del doc["calc_id"]
+
+        # Magnetism
+        doc["spin_polarized"] = "magnetism" in doc
+
+        doc["has_props"] = list(set(doc["has_props"]))
+
+        return SearchDoc(material_id=material_id, **doc)
+
+
+# Key mapping
+search_fields: Dict[str, list] = {
+    "materials": [
+        "nsites",
+        "elements",
+        "nelements",
+        "composition",
+        "composition_reduced",
+        "formula_pretty",
+        "formula_anonymous",
+        "chemsys",
+        "volume",
+        "density",
+        "density_atomic",
+        "symmetry",
+        "structure",
+        "deprecated",
+    ],
+    "thermo": [
+        "uncorrected_energy_per_atom",
+        "energy_per_atom",
+        "formation_energy_per_atom",
+        "energy_above_hull",
+        "is_stable",
+        "equillibrium_reaction_energy_per_atom",
+        "decomposes_to",
+    ],
+    "xas": ["absorbing_element", "edge", "spectrum_type", "xas_id"],
+    "grain_boundaries": [
+        "gb_energy",
+        "sigma",
+        "type",
+        "rotation_angle",
+        "w_sep",
+    ],
+    "electronic_structure": [
+        "band_gap",
+        "efermi",
+        "cbm",
+        "vbm",
+        "is_gap_direct",
+        "is_metal",
+        "bandstructure",
+        "dos",
+        "calc_id",
```

Comment:
`calc_id` should be switched back to `task_id` to match `ElectronicStructureBaseData`

Reply:
The only relevant line in the old builder was: `                    d[id]["es_source_calc_id"] = doc["calc_id"]` Was this a bug then?